src/ext/java/org/opentripplanner/ext/fares/impl/OrcaFareService.java

```diff
@@ -217,62 +246,65 @@ private static RideType classify(Route route, String tripId) {
   }
 
   /**
-   * Define which discount fare should be applied based on the fare type. If the ride type is unknown the discount
-   * fare can not be applied, use the default fare.
+   * Define which discount fare should be applied based on the fare type. If the ride type is
+   * unknown the discount fare can not be applied, use the default fare.
    */
-  private Money getLegFare(FareType fareType, RideType rideType, Money defaultFare, Leg leg) {
+  private Optional<Money> getLegFare(
+    FareType fareType,
+    RideType rideType,
+    Money defaultFare,
+    Leg leg
+  ) {
     if (rideType == null) {
-      return defaultFare;
+      return Optional.of(defaultFare);
+    }
+    // Filter out agencies that don't accept ORCA from the electronic fare type
+    if (usesOrca(fareType) && !agencyAcceptsOrca(rideType)) {
+      return Optional.empty();
     }
     return switch (fareType) {
-      case youth, electronicYouth -> getYouthFare();
+      case youth, electronicYouth -> Optional.of(getYouthFare());
       case electronicSpecial -> getLiftFare(rideType, defaultFare, leg.getRoute());
-      case electronicSenior, senior -> getSeniorFare(
-        fareType,
-        rideType,
-        defaultFare,
-        leg.getRoute()
-      );
+      case electronicSenior, senior -> getSeniorFare(fareType, rideType, defaultFare, leg);
       case regular, electronicRegular -> getRegularFare(fareType, rideType, defaultFare, leg);
-      default -> defaultFare;
+      default -> Optional.of(defaultFare);
     };
   }
 
+  private static Optional<Money> optionalUSD(float amount) {
+    return Optional.of(usDollars(amount));
+  }
+
   /**
    * Apply regular discount fares. If the ride type cannot be matched the default fare is used.
    */
-  private Money getRegularFare(FareType fareType, RideType rideType, Money defaultFare, Leg leg) {
+  private Optional<Money> getRegularFare(
```

Comment:
This method returns an `Optional` now but it appears to never return `Optional.empty`. Am I right?

If so, then you should simply wrap the call site in `Optional.of`.

Reply:
Good point, I think actually I do want this to return empty in some cases. Good catch